test/unit/plugins/rest_spec.js

```diff
@@ -845,13 +1032,63 @@ describe("The Rest plugin", function () {
 		});
 
 		after(function () {
+			terminateInstancesStub.restore();
 			updateStub.restore();
 		});
 
-		it("throws a 500 error", function () {
-			expect(result.statusCode).to.equal(500);
+		it("leaves the state unchanged(terminating)", function () {
+			expect(result.state).to.equal("terminating");
 		});
+	});
+
+	describe("when the database and terminateInstances fail", function () {
 
+		var result;
+		var getStub;
+		var terminateInstancesStub;
+		var server     = new Hapi.Server();
+		var instances  = new InstanceAdapter();
+		var awsAdapter = new AwsAdapter();
+		before(function () {
+
+			return instances.createInstance()
+			.then(function (instance) {
+				getStub = Sinon.stub(instances, "getInstance").rejects(new Error("Connection to the database fails."));
+				terminateInstancesStub = Sinon.stub(awsAdapter, "terminateInstances")
+				.rejects(new Error("AWS Error"));
+
+				server.connection();
+				server.registerAsync({
+					register : Rest,
+					options  : {
+						instances  : instances,
+						awsAdapter : awsAdapter
+					}
+				});
+
+				var request = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
+				.payload({
+					ami      : instance.ami,
+					type     : instance.type,
+					uri      : instance.uri,
+					state    : "terminating",
+					revision : instance.revision
+				});
+				return request.inject(server);
+			})
+			.then(function (response) {
+				result = response.result;
+			});
+		});
+
+		after(function () {
+			terminateInstancesStub.restore();
+			getStub.restore();
+		});
+
```

Comment:
This is the case where it can update successfully the first time setting the `state` to `terminating`. `terminateInstance()` and the database fail after this so the `state` at the end should still be `terminating`